core/templates/dev/head/pages/email_dashboard/EmailDashboard.js

```diff
@@ -0,0 +1,121 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for oppia email dashboard page.
+ */
+
+oppia.controller('EmailDashboard', [
+  '$scope', '$http', function($scope, $http) {
+    var QUERY_DATA_URL = '/emaildashboarddatahandler';
+
+    // Store current cursor.
+    $scope.currentCursor = null;
```

Comment:
Hm, this is an interesting approach, but I want to suggest a different one that I think is better design-wise and that is more consistent with what we do in other places where we have paging.

In general, we treat cursors as only having relevance in the backend, and don't want to spread them all over the frontend codebase. So, this controller should maintain only the latest cursor, and nothing else.

In addition, the controller should have an array along the lines of $scope.queries, a constant $scope.PAGE_SIZE, and a variable $scope.currentPageIndex.

Initially, $scope.queries = [], $scope.PAGE_SIZE = 10, $scope.currentPageIndex = 0. We make our first fetch, and $scope.queries has (up to) 10 entries. If the backend returns a cursor we display a 'next' arrow/button, otherwise we disable it.

When the 'next'/'previous' arrows/buttons are clicked, $scope.currentPageIndex increments/decrements by 1, as needed. The exception is when the 'next' arrow is clicked and we're at the end of the $scope.queries array, in which case we make another fetch with our current cursor to add more entries to $scope.queries, and then go forward in the array. In addition, when a new query is created, it should be pushed into the front of the array (after the backend has accepted it).

That way, you have an array $scope.queries and an index into it, which I think is a better data structure to represent what's on the page. You might also find it useful to have a look at core/templates/dev/head/pages/profile/Profile.js for an example that implements the frontend part of this approach. Let me know if you have any questions about this!